File: Practica 5 y 6/java/Punto3D.java
import java.lang.Math;

public class Punto3D {
    private double x;
    private double y;
    private double z;

    public Punto3D(double vx, double vy, double vz){
        setX(vx);
        setY(vy);
        setZ(vz);
    }

    private void setX(double vx){
        x = vx;
    }

    private void setY(double vy){
        y = vy;
    }

    private void setZ(double vz){
        z = vz;
    }

    private double GetX(){
        return x;
    }

    private double getY(){
        return y;
    }

    private double getZ(){
        return z;
    }

    public double distanciaP(Punto3D otroPunto){
        return Math.sqrt(Math.pow(this.x - otroPunto.x, 2) + Math.pow(this.y - otroPunto.y, 2) + Math.pow(this.z - otroPunto.z, 2));
    }
}

File: Practica 5 y 6/python/Punto3D.py
class Punto3D:
    def __init__(self, vx, vy, vz):
        self.setX(vx)
        self.setY(vy)
        self.setZ(vz)

    def setX(self, vx):
        self.x = vx

    def setY(self, vy):
        self.y = vy

    def setZ(self, vz):
        self.z = vz

    def getX(self):
        return self.x

    def getY(self):
        return self.y

    def getZ(self):
        return self.z

    def distanciaP(self, otroPunto):
        return ((self.x - otroPunto.x) ** 2 + (self.y - otroPunto.y) ** 2 + (self.z - otroPunto.z) ** 2) ** 0.5

if __name__ == "__main__":
    print("En este programa vamos a calcular la distancia entre puntos.")
    puntos = []

    for i in range(10):
        print(f"Ingresa las coordenadas del punto {i + 1}:")
        x = float(input("X: "))
        y = float(input("Y: "))
        z = float(input("Z: "))
        puntos.append(Punto3D(x, y, z))

    menorDistancia = float('inf')
    punto1, punto2 = -1, -1

    for i in range(10):
        for j in range(i + 1, 10):
            distancia = puntos[i].distanciaP(puntos[j])
            if distancia < menorDistancia:
                menorDistancia = distancia
                punto1 = i
                punto2 = j

    if punto1 != -1 and punto2 != -1:
        print(f"La menor distancia es entre el punto {punto1 + 1} y el punto {punto2 + 1}")
        print(f"Distancia: {menorDistancia}")
    else:
        print("No se encontraron dos puntos para calcular la distancia.")

